packages/ra-core/src/form/sanitizeEmptyValues.ts

```diff
@@ -16,8 +16,16 @@ const sanitizeEmptyValues = (initialValues: object, values: object) => {
     if (!initialValues) return values;
     const initialValuesWithEmptyFields = Object.keys(initialValues).reduce(
         (acc, key) => {
-            if (values[key] instanceof Date || Array.isArray(values[key])) {
```

Comment:
What if `values` is null or undefined? is it possible? shouldn't there be an if for that?